clippy_lints/src/needless_pass_by_value.rs

```diff
@@ -146,19 +145,13 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for NeedlessPassByValue {
                                            format!("&{}", snippet(cx, input.span, "_")));
                     }
 
-                    // Suggests adding `*` to dereference the added reference.
+                    // Suggests adding `*` to dereference the added reference if needed.
```

Comment:
The primary suggestions also need to go into the `spans` `Vec`, otherwise a tool like rustfix won't be able to know that they belong together and you could end up applying one but not the other.